test/tests/amqp/test_amqp_poller_alert.py

```diff
@@ -0,0 +1,367 @@
+            sleep(30)
+            rest_data = fit_common.rackhdapi('/redfish/v1/Systems/')
+            if rest_data['json']['Members@odata.count'] == 0:
+                continue
+            node_collection = rest_data['json']['Members']
+            for computenode in node_collection:
+                nodeidurl = computenode['@odata.id']
+                api_data = fit_common.rackhdapi(nodeidurl)
+                if api_data['status'] > 399:
+                    break
+                if node_uuid == api_data['json']['UUID']:
+                    return True
+        logs.debug_3("Time out to find the node with uuid!")
+        return False
+
+    def _wait_amqp_message(self, timeout):
+        timecount = 0
+        while amqp_queue.empty() is True and timecount < timeout:
+            sleep(1)
+            timecount = timecount + 1
+        self.assertNotEquals(
+            timecount,
+            timeout,
+            "AMQP message receive timeout")
+
+    def _get_ipmi_poller_by_node(self, node_id, seltype):
+        monurl = "/api/2.0/nodes/" + str(node_id) + "/pollers"
+        mondata = fit_common.rackhdapi(url_cmd=monurl)
+        if mondata['status'] not in [200, 201, 202, 204]:
+            self.fail("Get status: {},  Failed to get pollers for node: {}".format(mondata['status'], node_id))
+        else:
+            pollers = mondata['json']
+            for poller in pollers:
+                if poller["config"]["command"] == seltype:
+                    return poller["id"]
+        self.fail("Fail to find out ipmi poller")
+
+    def amqp_callback(self, ch, method, properties, body):
+        logs.debug_3("Routing Key {0}:".format(method.routing_key))
+        logs.data_log.debug_3(body.__str__())
+        global amqp_queue, nodefound_id
+        amqp_queue.put(
+            [method.routing_key, fit_common.json.loads(body)["nodeId"], body])
+        nodefound_id = fit_common.json.loads(body)["nodeId"]
+
+    def check_skupack(self):
+        sku_installed = fit_common.rackhdapi('/api/2.0/skus')['json']
+        if len(sku_installed) < 2:
+            return False
+        else:
+            return True
+
+    def _process_message(
+            self,
+            action,
+            typeid,
+            nodeid,
+            messagetype,
+            severity,
+            amqp_message_body):
+        expected_key = messagetype + "." + action + \
+            "." + severity + "." + typeid + "." + nodeid
+        expected_payload = {
+            "type": messagetype,
+            "action": action,
+            "typeId": typeid,
+            "nodeId": nodeid,
+            "severity": severity,
+            "version": "1.0",
+            "createdAt": {}}
+        self._compare_message(
+            amqp_message_body,
+            expected_key,
+            expected_payload)
+
+    def _compare_message(self, amqpmessage, expected_key, expected_payload):
+        routing_key = amqpmessage[0]
+        amqp_body = amqpmessage[2]

```

Comment:
I recommend dumping the full received message/routing keys in a debug level in this routine. This will help for debugging when any of these failure conditions below are hit, since they each just print out a single item.  Something like: logs.debug('received ampq message %s", amqpmessage) or below when you check the amqp_body is json format.